src/modules/feature-modules/admin/pages/users/roles/role-change.component.ts

```diff
@@ -20,57 +14,61 @@ import { AdminUsersService, changeUserRoleDTO } from '../../../services/users.se
 })
 export class PageUsersRoleChangeComponent extends CoreComponent implements OnInit {
 
-  user: { id: string, name: string };
-
-  organisation: null | UserInfo['roles'][0]['organisation'];
-
-  role: null | AccessorOrganisationRoleEnum;
-
-  roleName: string | null;
+  user: (UserInfo & { rolesDescription: string[], innovations?: GetInnovationsByOwnerIdDTO }) = { id: '', email: '', name: '', isActive: false, roles: [], rolesDescription: [] };
 
-  pageStep: 'RULES_LIST' | 'CODE_REQUEST' | 'SUCCESS' = 'RULES_LIST';
+  currentRole: string = '';
+  newRole: string = '';
 
-  rulesList: getOrganisationRoleRulesOutDTO[] = [];
+  userHasActiveRoles: boolean = false;
+  userHasInactiveRoles: boolean = false;
 
-  securityConfirmation = { id: '', code: '' };
-
-  form = new FormGroup({
-    code: new UntypedFormControl('')
-  }, { updateOn: 'blur' });
+  submitButton = { isActive: true, label: 'Change role' };
 
   constructor(
     private activatedRoute: ActivatedRoute,
-    private usersService: AdminUsersService,
-    private usersValidationRulesService: UsersValidationRulesService
+    private usersService: AdminUsersService
   ) {
 
     super();
-    this.user = {
-      id: this.activatedRoute.snapshot.params.userId,
-      name: RoutingHelper.getRouteData<any>(this.activatedRoute).user.displayName,
-    };
-    this.roleName = null;
-    this.role = null;
-    this.organisation = null;
+
+    this.user.id = this.activatedRoute.snapshot.params.userId;
+
   }
 
   ngOnInit(): void {
 
-    forkJoin([
-      this.usersService.getUserInfo(this.user.id),
-      this.usersValidationRulesService.getUserRoleRules(this.user.id)
-    ]).subscribe({
-      next: ([user, rules]) => {
+    this.usersService.getUserInfo(this.user.id).subscribe({
+      next: userInfo => {
+
+        userInfo.roles = userInfo.roles.filter(r => r.role === UserRoleEnum.ACCESSOR || r.role === UserRoleEnum.QUALIFYING_ACCESSOR);
+
+        if(userInfo.roles.filter(r => r.isActive).length === 0){
```

Comment:
It could be better to do `!userInfo.roles.some(r => r.isActive)`, that way you don't have always to traverse the entire array.